ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader.cpp

```diff
@@ -12,10 +16,33 @@ DdogCancellationTokenDeleter::operator()(ddog_CancellationToken* ptr) const
     }
 }
 
-Datadog::Uploader::Uploader(std::string_view _url, ddog_prof_Exporter* _ddog_exporter)
-  : url{ _url }
+Datadog::Uploader::Uploader(std::string_view _output_filename, ddog_prof_Exporter* _ddog_exporter)
+  : output_filename{ _output_filename }
   , ddog_exporter{ _ddog_exporter }
 {
+    // Increment the upload sequence number every time we build an uploader.
+    // Upoloaders are use-once-and-destroy.
+    upload_seq++;
+}
+
+bool
+Datadog::Uploader::export_to_file(ddog_prof_EncodedProfile* encoded)
+{
+    // Write the profile to a file using the following format for filename:
+    // <output_filename>.<process_id>.<sequence_number>
+    std::ostringstream oss;
+    oss << output_filename << "." << getpid() << "." << upload_seq;
```

Comment:
Good idea.  I was originally going to suggest that runtime_id should go here, but none of our testing frameworks care about runtime_id--they only know about the PID.

Reply:
Yeah, the current behavior is printing out pid by calling `os.getpid()`.